Assets/HoloToolkit/Utilities/Scripts/Singleton.cs

```diff
@@ -47,20 +57,29 @@ public static bool IsInitialized
         }
 
         /// <summary>
-        /// Base awake method that sets the singleton's unique instance.
+        /// Base Awake method that sets the Singleton's unique instance.
+        /// Called by Unity when initializing a MonoBehaviour.
+        /// Scripts that extend Singleton should be sure to call base.Awake() to ensure the
+        /// static Instance reference is properly created.
         /// </summary>
         protected virtual void Awake()
         {
-            if (instance != null)
+            if (IsInitialized && instance != this)
```

Comment:
I've seen other Singleton implementations call `DestroyImmediate(this)` in this case.

Reply:
`DestroyImmediate` is for in Editor where `Destory` is prefered at runtime.

Reply:
Added the auto destruct if an instance is found.